aten/src/ATen/native/native_functions.yaml

```diff
@@ -4575,6 +4572,8 @@
 
 - func: _softmax_backward_data(Tensor grad_output, Tensor output, int dim, ScalarType input_dtype) -> Tensor
   structured_delegate: _softmax_backward_data.out
+  dispatch:
+    NestedTensorCPU, NestedTensorCUDA: nested_softmax_backward
```

Comment:
bringing this comment from Alban over here:
```
You need to leave the CompositeImplicitAutograd key here
```

Reply:
Still can't get this to work if I add the CompositeImplicitAutograd key. I think because it is a structured delegate the dispatch table is in fact populated from the kernels defined from here:
```
- func: _softmax_backward_data.out(Tensor grad_output, Tensor output, int dim, ScalarType input_dtype, *, Tensor(a!) grad_input) -> Tensor(a!)
  structured: True
  dispatch:
    CPU: softmax_backward_cpu_out
    CUDA: softmax_backward_cuda_out
    MPS: softmax_backward_mps_out
```
so there is no `_softmax_backward_data` compositeImpllict kernel? or at least I am not sure what the name of that composite kernel is if it does exist, because it does not match the name `_softmax_backward_data`

Reply:
I looked at the build directory and I do see `_softmax_backward_data` registered to the CPU, CUDA,
```
m.impl("_softmax_backward_data", TORCH_FN(wrapper__softmax_backward_data));
```
can't find a RegisterMPS.cpp function in build so I am just assuming that it is also getting registered there as well

Reply:
@bdhirsh Does my above reasoning for why I get "undefined symbol" errors when trying to register composite implicit kernel for `_softmax_backward_data` fall in line with what you know about structured delegates?

Reply:
> I feel like structured kernels should never be structured as that is obviously not composite compliant?

What exactly is the problem here? Is something bypassing dispatch?

Reply:
Not to derail the convo but is the main issue here that the softmax_kernel for nested tensors was only registered/written for the functional version when in fact it should have been written for the out variant?

Reply:
> Not to derail the convo but is the main issue here that the softmax_kernel for nested tensors was only registered/written for the functional version when in fact it should have been written for the out variant?

This seems right to me - I'd expect the functional version to be delegated to the out variant.

Reply:
After the sage wisdom from Brian for native functions with a `structured_delegate` key. A `CompositeExplicitAutogradNonFunctional` kernel gets registered regardless of dispatch entries therefore it so not required to manual add here.